### PR TITLE
Make symbolicate_event blocking and remove retry from sleep queue

### PR DESCRIPTION
**Pull request by beezz**
Making `symbolicate_event` a blocking task, without scheduling retries and result querying on the sleep queue, gives us the ability to better control the maximum number of concurrent tasks submitted to Symbolicator and prevent its overload (but rather backlogging symbolicate_event queue).

I suggest we, rather significantly, increase the initial polling time 4s -> 30s which should be sufficient for more than 95% of symbolications.

If we want to have a quick way to reverse back to the sleep queue behavior we can introduce an option that would switch between the blocking and non-blocking implementation of the `_do_symbolicate_event` but with the current capacity of the `symbolicate_event` workers, I don't think that's necessary.

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -79,7 +83,7 @@ def submit_process(
 
 
 def sample_symbolicate_event_apm():
-    return random.random() < getattr(settings, "SENTRY_SYMBOLICATE_EVENT_APM_SAMPLING", 0)
+    return random.random() < SYMBOLICATE_EVENT_APM_SAMPLING
```
Comment (beezz) [inline src/sentry/tasks/store.py:86]: Unrelated change. Just resolve the sampling rate once on the module level.

File: src/sentry/lang/native/symbolicator.py [hunk 2]
```diff
@@ -142,7 +142,7 @@ def _process(self, create_task):
                 # retrying after just a second.
                 #
                 # If there is no response attached, it's a connection error.
-                raise RetrySymbolication(retry_after=10)
+                raise RetrySymbolication(retry_after=3)
```
Comment (beezz) [inline src/sentry/lang/native/symbolicator.py:145]: 10 seconds seems a bit too much, I'd rather see us retry more often even in rare 502/503 cases.

File: src/sentry/tasks/store.py [hunk 3]
```diff
@@ -210,79 +214,76 @@ def _do_symbolicate_event(cache_key, start_time, event_id, symbolicate_task, dat
...
+                        data.setdefault("_metrics", {})["flag.processing.fatal"] = True
+                        has_changed = True
+                        break
+                    else:
+                        # sleep for `retry_after` seconds and try again
+                        metrics.incr(
+                            "tasks.store.symbolicate_event.retry",
+                            tags={"symbolication_function": symbolication_function.__name__},
+                        )
+                        sleep(e.retry_after)
```
Comment (beezz) [inline src/sentry/tasks/store.py:272]: I'm not sure it's a good idea to always respect the `retry_after` atm it seems to be hardcoded in symbolicator to 30s. https://github.com/getsentry/symbolicator/blob/5fd8b3d4c0b5299fecd5864e171b34df7ba80c4c/src/actors/symbolication.rs#L234

Comment (beezz) [reply src/sentry/tasks/store.py]: Introduced a new setting that will max it to 5s regardless of what comes from symbolicator. This might be revisited once symbolicator implements more sophisticated estimations.

File: src/sentry/tasks/store.py [hunk 4]
```diff
@@ -307,8 +310,9 @@ def _do_symbolicate_event(cache_key, start_time, event_id, symbolicate_task, dat
 @instrumented_task(
     name="sentry.tasks.store.symbolicate_event",
     queue="events.symbolicate_event",
-    time_limit=65,
-    soft_time_limit=60,
+    time_limit=settings.SYMBOLICATOR_PROCESS_EVENT_HARD_TIMEOUT + 30,
+    soft_time_limit=settings.SYMBOLICATOR_PROCESS_EVENT_HARD_TIMEOUT + 20,
+    acks_late=True,
```
Comment (beezz) [inline src/sentry/tasks/store.py:315]: `acks_late=True` means the task will be acked only after completion and tasks killed during deploy will be re-enqueued.

**Review by tonyo — APPROVED** [review]

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (jan-auer) [reply src/sentry/lang/native/symbolicator.py:145]: Can this just be `SYMBOLICATOR_MAX_RETRY_AFTER`?

File: src/sentry/tasks/store.py [hunk 5]
```diff
@@ -31,6 +31,12 @@
 # Is reprocessing on or off by default?
 REPROCESSING_DEFAULT = False
 
+SYMBOLICATE_EVENT_APM_SAMPLING = float(
+    getattr(settings, "SENTRY_SYMBOLICATE_EVENT_APM_SAMPLING", 0.0)
+)
+
+SYMBOLICATOR_MAX_RETRY_AFTER = float(getattr(settings, "SYMBOLICATOR_MAX_RETRY_AFTER", 5))
```
Comment (jan-auer) [inline src/sentry/tasks/store.py:38]: Why are we not using `settings.SYMBOLICATOR_MAX_RETRY_AFTER` directly but going through `getattr`?

**Review by jan-auer — APPROVED** [review]

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (beezz) [reply src/sentry/lang/native/symbolicator.py:145]: Yeah, I introduced the `SYMBOLICATOR_MAX_RETRY_AFTER` only after this change, so this is a good idea.

File: src/sentry/tasks/store.py (hunk 5 above)
Comment (beezz) [reply src/sentry/tasks/store.py]: Just because it's like this all over the place and we are not consistent.

I believe the idea is that someone might be missing the setting and with `gettattr` and default value we prevent failures in such a case but imo it is safe here as we ship the defaults in `server.py`.
